src/client/components/Editor/EditorBase.js

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { isValidImage } from '../../helpers/image';
+
+function getDisplayName(WrappedComponent) {
+  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+}
+
+export default function editorBase(WrappedComponent) {
+  class editorBaseComponent extends React.Component {
+    static propTypes = {
+      onImageInserted: PropTypes.func,
+      onImageInvalid: PropTypes.func,
+    };
+
+    static defaultProps = {
+      onImageInserted: () => {},
+      onImageInvalid: () => {},
+    }
+
+    state = {
+      isDisabledSubmit: false,
+      imageUploading: false,
+      dropzoneActive: false,
+    };
+
+    updateIsDisabledSubmit = (isDisabledSubmit) => {
```

Comment:
I think we should bind all these in the constructor instead of using arrow functions from now on.

https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1

Reply:
I can't see to figure out how, could you help @jm90m

Reply:
Nevermind, I figured it out.